src/Cli/dotnet/commands/dotnet-workload/WorkloadHistoryRecorder.cs

```diff
@@ -54,15 +54,16 @@ public void Run(Action workloadAction)
         private WorkloadHistoryState GetWorkloadState()
         {
             var resolver = _workloadResolverFunc();
-            var currentWorkloadInfo = resolver.GetWorkloadVersion();
+            var currentWorkloadVersion = resolver.GetWorkloadVersion();
+            currentWorkloadVersion = currentWorkloadVersion.Contains("manifest") ? string.Empty : currentWorkloadVersion;
```

Comment:
Is there an api that tells you you're in workload set (or not) mode that you could use instead of doing the string check?

Reply:
It sounded like there was consensus behind this, so I implemented it. I want to point out, however, that it made the PR roughly 25x as big...albeit from 'tiny' to 'small'